haskell/toolchain.bzl

```diff
@@ -27,21 +27,39 @@ load(
 
 _GHC_BINARIES = ["ghc", "ghc-pkg", "hsc2hs", "haddock", "ghci", "runghc", "hpc"]
 
-def _run_ghc(hs, cc, inputs, outputs, mnemonic, arguments, env, params_file = None, progress_message = None, input_manifests = None, extra_name = ""):
+def _run_ghc(
+        hs,
+        cc,
+        inputs,
+        outputs,
+        mnemonic,
+        arguments,
+        env,
+        params_file = None,
+        progress_message = None,
+        input_manifests = None,
+        extra_name = "",
+        worker = None):
     args = hs.actions.args()
     extra_inputs = []
 
     # Detect persistent worker support
-    flagsfile_prefix = ""
     execution_requirements = {}
     tools = []
-    if hs.worker != None:
-        flagsfile_prefix = "@"
-        execution_requirements = {"supports-workers": "1"}
-        args.add(hs.worker.path)
-        tools = [hs.worker]
+    if hs.worker != None and worker == None:
+        worker = hs.worker
+
+    if worker != None:
+        execution_requirements = {
+            "supports-workers": "1",
+            "requires-worker-protocol": "proto",
+            "supports-multiplex-workers": "1",
+            "supports-multiplex-sandboxing": "1",
+        }
```

Comment:
I think this change in `execution_requirements` is disabling the old persistent worker for `haskell_library`. Should I make efforts to keep it functioning? (if it is still functioning condition, I haven't tried it).